dp/cloud/go/services/dp/storage/amc_manager_test.go

```diff
@@ -0,0 +1,652 @@
+						WithSingleStepEnabled(true).
+						WithFullInstallationParam().
+						WithCbsdCategory("a").
+						Cbsd,
+					registered,
+					registered).
+				Details,
+		},
+	}, {
+		name: "test get state with single step enabled",
+		input: []db.Model{
+			b.NewDBCbsdBuilder().
+				WithNetworkId(someNetwork).
+				WithSerialNumber(fmt.Sprintf(someSerialNumber+"%d", 11)).
+				WithId(11).
+				WithCbsdId(someCbsdIdStr).
+				WithStateId(registeredId).
+				WithDesiredStateId(registeredId).
+				WithSingleStepEnabled(true).
+				WithFullInstallationParam().
+				WithCbsdCategory("a").
+				Cbsd,
+		},
+		expected: []*storage.DetailedCbsd{
+			b.NewDetailedDBCbsdBuilder().
+				WithCbsd(
+					b.NewDBCbsdBuilder().
+						WithNetworkId(someNetwork).
+						WithSerialNumber(fmt.Sprintf(someSerialNumber+"%d", 11)).
+						WithCbsdId(someCbsdIdStr).
+						WithShouldDeregister(false).
+						WithShouldRelinquish(false).
+						WithIsDeleted(false).
+						WithSingleStepEnabled(true).
+						WithFullInstallationParam().
+						WithCbsdCategory("a").
+						Cbsd,
+					registered,
+					registered).
+				Details,
+		},
+	}, {
+		name: "test not get state without registration params",
+		input: []db.Model{
+			b.NewDBCbsdBuilder().
+				Empty().
+				WithId(12).
+				WithNetworkId(someNetwork).
+				WithPreferences(15, []int64{3600, 3580, 3620}).
+				WithStateId(registeredId).
+				WithMinPower(1).
+				WithMaxPower(2).
+				WithNumberOfPorts(2).
+				WithDesiredStateId(registeredId).
+				Cbsd,
+		},
+		expected: []*storage.DetailedCbsd{},
+	}, {
+		name: "test not get state without eirp capabilities",
+		input: []db.Model{
+			b.NewDBCbsdBuilder().
+				Empty().
+				WithId(13).
+				WithFccId(someFccId).
+				WithUserId(someUserId).
+				WithNetworkId(someNetwork).
+				WithCbsdId(someCbsdIdStr).
+				WithMinPower(1).
+				WithMaxPower(2).
+				WithNumberOfPorts(2).
+				WithSerialNumber(fmt.Sprintf(someSerialNumber+"%d", 13)).
+				WithPreferences(15, []int64{3600, 3580, 3620}).
+				WithStateId(registeredId).
+				WithDesiredStateId(registeredId).
+				Cbsd,
+		},
+		expected: []*storage.DetailedCbsd{},
+	}, {
+		name: "test not get state with single step enabled and no installation params",
+		input: []db.Model{
+			b.NewDBCbsdBuilder().
+				WithNetworkId(someNetwork).
+				WithSerialNumber(fmt.Sprintf(someSerialNumber+"%d", 14)).
+				WithFccId(someFccId).
+				WithUserId(someUserId).
+				WithNetworkId(someNetwork).
+				WithCbsdId(someCbsdIdStr).
+				WithStateId(registeredId).
+				WithMinPower(1).
+				WithMaxPower(2).
+				WithNumberOfPorts(2).
+				WithDesiredStateId(registeredId).
+				WithSingleStepEnabled(true).
+				WithCbsdCategory("a").
+				Cbsd,
+		},
+		expected: []*storage.DetailedCbsd{},
+	}, {
+		name: "test not get state with single step enabled, category A and outdoor",
+		input: []db.Model{
+			b.NewDBCbsdBuilder().
+				WithNetworkId(someNetwork).
+				WithSerialNumber(fmt.Sprintf(someSerialNumber+"%d", 14)).
```

Comment:
I would do this `fmt.Sprintf("%s%d", someSerialNumber, 14)` or this `someSerialNumber+"14"`, to avoid mixing different methods of building strings